bouncer/shared/fill_or_kill.ts

```diff
@@ -0,0 +1,139 @@
+import { InternalAsset as Asset, InternalAssets as Assets } from '@chainflip/cli';
+import assert from 'assert';
+import {
+  amountToFineAmount,
+  amountToFineAmountBigInt,
+  assetDecimals,
+  decodeDotAddressForContract,
+  isWithinOnePercent,
+  newAddress,
+  observeBalanceIncrease,
+  shortChainFromAsset,
+} from './utils';
+import { requestNewSwap } from './perform_swap';
+import { send } from './send';
+import { getBalance } from './get_balance';
+import { getChainflipApi, observeEvent } from './utils/substrate';
+import { RefundParameters } from './new_swap';
+
+/// Do a swap with unrealistic minimum price so it gets retried and then refunded.
+async function testMinPriceRefund(asset: Asset, amount: number) {
+  const swapAsset = asset === Assets.Usdc ? Assets.Flip : Assets.Usdc;
+  const refundAddress = await newAddress(asset, 'FOK_REFUND');
+  const destAddress = await newAddress(swapAsset, 'FOK');
+  console.log(`Swap destination address: ${destAddress}`);
+
+  const refundBalanceBefore = await getBalance(asset, refundAddress);
+
+  const invalidMinPrice = amountToFineAmount(
+    '99999999999999999999999999999999999999999999999999999',
+    assetDecimals(asset),
+  );
+
+  await using chainflip = await getChainflipApi();
+  const currentRetryDuration = Number(await chainflip.query.swapping.swapRetryDelay());
+  console.log(`Current swap retry delay: ${currentRetryDuration} blocks`);
+
+  const refundParameters: RefundParameters = {
+    retryDurationBlocks: 1.6 * currentRetryDuration, // More than the swap retry delay, so it will be retried once.
+    refundAddress:
+      asset === Assets.Dot ? decodeDotAddressForContract(refundAddress) : refundAddress,
+    minPrice: invalidMinPrice,
+  };
+
+  console.log(`Requesting swap from ${asset} to ${swapAsset} with unrealistic min price`);
```

Comment:
If we are going to run this test in parallel to others, we should tag all `console.log` statements (they way we do for other parallel tests) so we can link the message to the test.

Reply:
We don't do this anywhere else in the bouncer. Do you think its worth starting now? or should we come up with a better solution for logging in general? eg, something that can auto prefix.

Reply:
There is a solution to this i just found in the gaslimit_ccm.ts test. Starts building a list of events before the deposit and then continuously checks the list once we know the `swapRequestId`. Maybe we could put this solution into a nicer utility of some sort.